src/Roslyn.Diagnostics.Analyzers/Core/Extensions/INamedTypeSymbolExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis;
+
+namespace Roslyn.Diagnostics.Analyzers.Extensions
+{
+    internal static class INamedTypeSymbolExtensions
+    {
+        public static bool IsTestAttribute(this INamedTypeSymbol attributeClass, ConcurrentDictionary<INamedTypeSymbol, bool> knownTestAttributes, INamedTypeSymbol factAttribute)
```

Comment:
Should we extract the test attribute list from here into some helper type that can then be shared? https://github.com/dotnet/roslyn-analyzers/blob/f06e25d8ac47ae5a731107c7c82d9d09f343b8ba/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs#L293-L310

Reply:
Right now this suppressor is specific to dotnet/roslyn, which uses XUnit.

Reply:
I didn't evaluate the impact of all of these on the suppressor, so I focused on the primary known case.